src/colmap/scene/database.cc

```diff
@@ -453,6 +453,13 @@ PosePrior Database::ReadPosePrior(const image_t image_id) const {
         ReadStaticMatrixBlob<Eigen::Vector3d>(sql_stmt_read_pose_prior_, rc, 1);
     prior.coordinate_system = static_cast<PosePrior::CoordinateSystem>(
         sqlite3_column_int64(sql_stmt_read_pose_prior_, 2));
+    prior.position_covariance =
+        ReadStaticMatrixBlob<Eigen::Matrix3d>(sql_stmt_read_pose_prior_, rc, 3);
+
+    if (prior.position_covariance.isZero()) {
```

Comment:
I am curious why this is needed - if NaNs are written with `WriteStaticMatrixBlob`, isn't the matrix returned by `ReadStaticMatrixBlob` NaN too?

Reply:
I did this because when the position_covariance column is added to databases that were not created this new table structure, the covariance matrix is set to zero by default.

Reply:
You might be able to pass a default value when adding the column, something like this:
```c++
const std::string sql = "ALTER TABLE pose_priors ADD COLUMN position_covariance BLOB DEFAULT (?);";
sqlite3_stmt* sql_stmt;
SQLITE3_CALL(sqlite3_prepare_v2(database_, sql.c_str(), -1, &sql_stmt, 0));
WriteStaticMatrixBlob(sql_stmt, PosePrior().position_covariance, 1);
SQLITE3_CALL(sqlite3_step(sql_stmt));
SQLITE3_CALL(sqlite3_finalize(sql_stmt));
```

Reply:
Done :+1: 

It seems it is not possible to set a dynamic default parameter but I solved it by directly applying an update procedure on the column.